src/routes/image.rs

```diff
@@ -97,6 +99,10 @@ impl IntoResponse for ImageProcessingError {
                 StatusCode::BAD_REQUEST,
                 format!("The provided resource URI is not valid: '{}'", resource_uri)
             ),
+            ImageProcessingError::FileSizeExceeded => (
+                StatusCode::BAD_REQUEST,
+                String::from("The image exceeds the allowed size. Please ensure the file size is within the permissible limit."),
```

Comment:
I would add the `maximum allowed size` instead of just `allowed size` and actually display the maximum allowed size like this:
```rust
ImageProcessingError::FileSizeExceeded(max_allowed_size) => (
                StatusCode::BAD_REQUEST,
                format!("The image exceeds the allowed size of '{max_allowed_size}'. Please ensure the file size is within the permissible limit or adjust the configuration."),
            )
```